Tasks/Common/ServiceFabricHelpers/Connect-ServiceFabricClusterFromServiceEndpoint.ps1

```diff
@@ -103,8 +103,16 @@ function Add-Certificate
     Write-Host (Get-VstsLocString -Key ImportedCertificate -ArgumentList $certificate.Thumbprint)
 
     # Override the certificate-related cluster connection parameters to known and supported values
-    $clusterConnectionParameters["FindType"] = "FindByThumbprint"
-    $clusterConnectionParameters["FindValue"] = $certificate.Thumbprint
+    if ($ConnectedServiceEndpoint.Auth.Parameters.ServerCertCommonName) 
+    {
+        $clusterConnectionParameters["FindType"] = "FindBySubjectName"
```

Comment:
I am not sure why this is needed. This is for client certificate. We always find client certificate by thumbprint only. The option for common name is for server certificate which is already being handled in L197

Reply:
According to the doc: https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-connect-to-secure-cluster
We should be using "FindBySubjectName" for finding certificates with Common Name

Reply:
Common name should be same for client and server certificate, for Service Connection to connect using common name

Reply:
Both Server cert as well as client cert needs to have Common Name. They can be same or different. The input in Service Connection is for Server cert common name. But the ```Connect-ServiceFabricCluster``` command we use needs common name for Client cert as well as per the doc

Reply:
It does not say that you need to use common name for both certs. We can choose to use common name for server cert and thumbprint for client cert. I am not completely convinced that this would cause intermittent failure.
Do you have a repro? And does it get fixed by this fix? That would convince.
We can talk to service fabric team if we want to confirm this behavior/issue.